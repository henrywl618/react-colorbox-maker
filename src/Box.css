.Box{
    display: inline-block;
    margin: 2px;
}

.Box-button{
    position: relative;
    left: -28px;
    bottom: 10px;
}