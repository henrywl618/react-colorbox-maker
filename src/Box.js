import React from "react";
import "./Box.css";

const Box = ({width,height,color,id, removeBox})=>{
    const boxStyle = {
        width: width,
        height: height,
        backgroundColor: color,
        id: id,
    };
    return (
    <>
        <div key={id} className="Box" style={boxStyle}></div>
        <button className="Box-button" onClick={removeBox}>X</button>

    </>
    )
};

export default Box;