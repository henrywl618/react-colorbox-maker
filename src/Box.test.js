import React from "react";
import {render} from "@testing-library/react";
import Box from "./Box";

it("renders without crashing", ()=>{
    render(<Box width='100px' height='100px' color='black' />);
});

it("matches the snapshot", ()=>{
    const { asFragment } =  render(<Box width='100px' height='100px' color='black' />);
    expect(asFragment()).toMatchSnapshot();
});