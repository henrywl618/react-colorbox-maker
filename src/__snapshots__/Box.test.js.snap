// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 100px; height: 100px; background-color: black;"
  />
  <button
    class="Box-button"
  >
    X
  </button>
</DocumentFragment>
`;
